_posts/2020-10-30-snowflake-dataOps.md

```diff
@@ -0,0 +1,154 @@
+---
+layout: post
+title: DataOps with AWS and Snowflake
+date: 2020-10-27
+tags: devops, dataops, aws, snowflake, ci, cd, flyway
+author: Zainab Maleki
+image: img/blog/dataops/dataOps.jpg
+---
+
+<center><img src="/img/blog/dataops/dataOps.jpg" /></center><br/>
+
+# What is DataOps?
+
+DevOps practices have been discovered, utilised and perfected over the past few years to allow speed and agility in software engineering. Despite these advancements in the software engineering discipline, they have not been widely adopted in the data engineering domain. DataOps is a concept that applies DevOps practices to data life cycles to achieve speed and agility whilst delivering trusted and high-quality data. Like DevOps, DataOps minimises manual interventions by codifying data related changes and deployment through automated CI/CD tools.<br/><br/>
+In Snowflake just about everything can be managed via executable SQL statements. This makes schema migration tools like [Flyway](https://flywaydb.org/documentation/database/snowflake){:target="_blank"} and [Sqitch](https://sqitch.org/docs/manual/sqitchtutorial-snowflake/){:target="_blank"} the perfect devops companion for Snowflake. The schema migration tools are great for creating and managing resources, like grants, databases, snowpipes, stages, tasks, integrations etc. While it is possible to also manage data transformations using flyway (e.g. via tasks), this is a job better suited to tools like dbt.<br/>
+
+In this blog post, I will show you how to create your Snowflake DataOps pipeline using AWS developer tools and Flyway.
+
+<div style="background-color: #fff3cd ; border-color: #ffeeba; color: #856404; border-radius: .25rem; padding: .75rem 1.25rem;"><strong>Warning!</strong><br/>This is a very hands-on blog post. Get ready to get your hands dirty &#128525;</div> <br/>
+
+## Let's get Started:
+
+To get started you will need to follow below steps:
+
+# Step 1: Downloading the template
+Download below repository as a zip folder. Once downloaded unzip it and cd into the folder<br/>
+[Snowflake Dataops Repository](https://github.com/MechanicalRock/snowflake-dataops){:target="_blank"}
+
+This repo uses below services to implement an automated deployment cycle to Snowflake. All the pipeline and related infrastructure are created through code using AWS cloudformation.
+
+* CodeCommit
+* CodeBuild
+* CodePipeline
+* Cloudformation
+* Flyway
+
+# Step 2: Setup Snowflake Credentials
+
+To allow your pipeline to get access to Snowflake, you will need to first create an RSA public and private key. You can create the keys using openssl by running below commands in your command line
+```
```

Comment:
tip; you can get language specific code formatting by putting the language code after the three ticks.

e.g. \`\`\`sql and \`\`\`sh